src/Search.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from "react";
+import { ReactSearchAutocomplete } from "react-search-autocomplete";
+import "./search.css";
+
+const autocompleteURL =
+  "https://autocomplete.search.hereapi.com/v1/autocomplete?";
+
+function autoCompleteCity(city) {
+  if (!city) return Promise.resolve("");
+
+  const query = `q=${city}&limit=4&types=city&apiKey=${process.env.REACT_APP_HEREAPI}`;
+  return fetch(`${autocompleteURL}${query}`)
+    .then((res) => res.json())
+    .then((result) => {
+      console.log(result);
+      if(!result.items) return []
+      const set = result.items.map(
+        (item) => `${item.address.city}, ${item.address.countryName}`
+      );
+      const uniqueCities = [...new Set(set)];
+
+      const filterdCities = uniqueCities.filter(function(x) {
+        return x !== undefined;
+      });
+      let newitems = [];
+      let count = 0;
+      filterdCities.forEach((city) => {
+        newitems.push({ id: count, name: city });
+        count += 1;
+      });
+
+      return newitems;
+    });
+}
+
+export default function SearchBox({ setCity }) {
+  const [searchText, setSearchText] = React.useState("");
+  const [items, setItems] = React.useState([]);
+
+  const _handleOnSearch = async (searchText) => {
+    const items = await autoCompleteCity(searchText);
+    setItems(items);
+    setSearchText(searchText);
+  };
+
+  const _handleOnSelect = ({ name }) => {
+    setSearchText(name);
+    setCity(name);
```

Comment:
![image](https://user-images.githubusercontent.com/86974918/180426632-b48b21d8-c979-4e03-ba02-d11e24336291.png)
For some cities selected from the list, the weather API gives 404 not found error. Could you please just add a check that if the city gives the code 404 just display a message 'weather not found' or something.

Reply:
I guess that is to be added in the weather part?

Reply:
Yup maybe we can create a new issue to handle that.